plugins/modules/network_settings_workflow_manager.py

```diff
@@ -1043,167 +1318,316 @@ def get_network_params(self, site_id):
             suitable for Cisco Catalyst Center configuration, or None
             if the response is not a dictionary or there was an error.
         """
-
-        try:
+        if self.get_ccc_version_as_integer() <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
```

Comment:
Lot more code in this API. Can we split into two APIs?
get_network_params_v1 
get_network_params_v2
```

if self.get_ccc_version_as_integer() <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
    self.get_network_params_v1 (site_name, site_id, any args that are needed).check_return_status()
else:
    self.get_network_params_v2(site_name, site_id, any args that are needed)).check_return_status()
```

Do common things here and return network_details

Reply:
Done the changes.